### ISSUE TITLE
Allow component class to the tagName prop

### ISSUE THREAD

**Issue by rokoroku** [yahoo/react-intl#1168]
### Desired behavior

```jsx
const Header = (props) => <h1 {...props}/>

...

render () {
  return <FormattedMessage tagName={Header} />
}
```

### Current behavior
It works, however, it emits react warning since the propType for `tagName` is set to `string`. 

### Your Environment
| Executable | Version |
| ---: | :--- |
| `npm ls react-intl` | 2.4.0 |
| `npm ls react` | 16.4.1 |

**Comment by papasmile** [comment]
Hi, @rokoroku, sorry it's not totally what you are trying to do but there are several ways to use FormattedMessage (I think that's what you mean not Formatted*String*): https://github.com/yahoo/react-intl/wiki/Components#formattedmessage

e.g.
```
<FormattedString tagName="h1" />
```
or if you want your `props` on your h1:
```
<FormattedMessage id="title">
    {(txt) => (
        <Header {...props}/>
            {txt}
        </Header>
    )}
</FormattedMessage>
```

Hope that helps!

**Comment by rokoroku** [comment]
Oops, sorry, I meant `tagName` prop of `<FormattedMessage />`.

Yeah of course we can use function as children, but IMO we can also use tagName as a component just like what IntlProvider do: `<IntlProvider textComponent={Text}/>`

And it's possible since FormattedMessage already use the `React.createElement` method inside its render method: 
https://github.com/yahoo/react-intl/blob/763f3fcbf87abc1a2a1021f684217f56bf8d4581/src/components/message.js#L135

**Comment by papasmile** [comment]
Ah, so I think you are saying that your example actually works as you expect it (message is wrapped in Header) but it just gives a props error because it is marked as 'string', right?

**Comment by rokoroku** [comment]
Yup, I've just corrected the content.

**Comment by rokoroku** [comment]
We can just change the `propType` and API documentation to resolve this issue, if it doesn't ruin its intention (allowing only string tagName)

If approved, I'd happy to write a PR for this :)

**Comment by lexanth** [comment]
Should this be `tagName` or leave that as a string and have a new prop e.g. `component`?

**Comment by lexanth** [comment]
Also the prop type for a Fragment is quite tricky as Fragments are just symbols. Could use PropTypes.symbol, but then it's not specific to Fragment. The fragment symbol is specific to the version of React in use - e.g. https://github.com/erikras/redux-form/issues/4141 

Or just (untested):
```
PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.function, // Covers classes and SFCs
  PropTypes.oneOf([Fragment]) // Not sure if this works
])
```

**Comment by sourabh2k15** [comment]
since this PR is merged -> #1118 we could close this issue

**Comment by rokoroku** [comment]
👍  for #1118, closing
